src/components/Email.js

```diff
@@ -35,19 +36,30 @@ const Component = (props) => {
     //    nativeValidation: true,
     defaultValues: data,
   });
-  const { watch, getValues, setError, clearErrors } = form;
+  const { watch, getValues, setValue, setError, clearErrors } = form;
 
   const country = watch ("country");
   const fields = getValues (["subject","message"]);
 
+  const tokenKeys = extractTokens (data["message"]);
+  const tokens = watch (tokenKeys);
+  const handleMerging = text => {
+    setValue("message",text);
+  };
+
+  if (tokenKeys.includes ("targets"))
+    tokens.targets=profiles;
+  const merged = useToken (data["message"],tokens, handleMerging);
```

Comment:
merged gets the merged text, but also in handleMerging, message is set to merged text. So there are two channels to pass this merged text up, I guess just using useToken return value is enough?

Reply:
it did trigger endless loop if I started to update the message there ;( #hackhackhack